aten/src/ATen/native/cpu/PowKernel.cpp

```diff
@@ -35,66 +36,95 @@ void pow_tensor_tensor_kernel(TensorIterator& iter) {
 }
 
 void pow_tensor_scalar_kernel(TensorIterator& iter, const Scalar& exp_scalar) {
-  if (isFloatingType(iter.dtype())) {
-    const auto exp = exp_scalar.to<double>();
-    // Floating types allow AVX2 vector optimizations for pow/sqrt/rsqrt:
-    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "pow", [&]() {
-      using Vec = Vec256<scalar_t>;
-      if (exp == 0.5) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return std::sqrt(base);
-          },
-          [](Vec base) -> Vec { return base.sqrt(); }
-        );
-      } else if (exp == 2) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return base * base;
-          },
-          [](Vec base) -> Vec { return base * base; }
-        );
-      } else if (exp == 3) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return base * base * base;
-          },
-          [](Vec base) -> Vec { return base * base * base; }
-        );
-      } else if (exp == -0.5) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) __ubsan_ignore_float_divide_by_zero__ -> scalar_t {
-            return 1.0 / std::sqrt(base);
-          },
-          [](Vec base) -> Vec { return base.rsqrt(); }
-        );
-      } else if (exp == -1) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return 1.0 / base;
-          },
-          [](Vec base) -> Vec { return base.reciprocal(); }
-        );
-      } else if (exp == -2) {
+  // prevent multiple calls to iter.common_dtype()
```

Comment:
Hey @imaginary-person, this looks good but do you mind if I tweak it a little. I have an idea that I'd like to try here.

Reply:
Of course I wouldn't mind at all :)

Reply:
@heitorschueroff, if you'd be reviewing other files in this PR as well, can you please take a look at [the changes I made](https://github.com/pytorch/pytorch/commit/d21da15c4612a33dcb0b8349cf56ba80b34ddc8f#diff-c99ae5b91314e5551d6b61d1d91539c550b844cfe38a7a063c44ae4d2b479c15) to `test_int_pow` in `test_binary_ufuncs.py'? It'd be great for me to get some feedback before refactoring/adding similar tests for other dtypes.

That particular test only tests for integral base & exponent, but I'd add tests for exponents of other dtypes as well.
Thank you!

Reply:
BTW @heitorschueroff, did you want me to mark the PR as `Ready for review` so that you could use the `Suggested changes` feature? I'm not sure if a reviewer can do so while the PR is in draft stage, so I'll mark it as `Ready for Review`. Thanks!

Reply:
> BTW @heitorschueroff, did you want me to mark the PR as `Ready for review` so that you could use the `Suggest changes` feature? I'm not sure if a reviewer can do so while the PR is in draft state, so I'll mark it as `Ready for Review`. Thanks!
> 
> Update: If you have write access to the Pytorch repo, then you also have write access to my repo that I used to create this PR. It seems that the state of the PR doesn't matter with regards to write access.

Sorry for the late reply but yes, reviewers can use the suggest changes even in a draft PR. The only different between a regular and a draft PR is that draft PRs don't automatically notify people about them.

Reply:
Thanks for your response! 
BTW, I'll rebase this PR as well, as currently, it has merge-conflicts.